src/core/server/elasticsearch/elasticsearch_service.ts

```diff
@@ -94,11 +95,67 @@ export class ElasticsearchService implements CoreService<InternalElasticsearchSe
 
     const config = await this.config$.pipe(first()).toPromise();
 
+    const adminClient$ = clients$.pipe(map(clients => clients.adminClient));
+    const dataClient$ = clients$.pipe(map(clients => clients.dataClient));
+
+    const adminClient = {
+      async callAsInternalUser(
+        endpoint: string,
+        clientParams: Record<string, any> = {},
+        options?: CallAPIOptions
+      ) {
+        const client = await adminClient$.pipe(take(1)).toPromise();
```

Comment:
shouldn't we use `last()` instead of `take(1)`, if more than one configuration change occurred since the previous `callAsInternalUser` call we would get a stale client here.